browser/js/common/directives/graphs/add-graph.directive.js

```diff
@@ -0,0 +1,77 @@
+app.directive('addGraph', function($http) {
```

Comment:
If you're injecting `$http` into a directive, it's probably a good indication that there should be some factory somewhere that abstracts away the whole talking-to-the-backend stuff. And in general, I'm noticing that this directive has some moderate code size—another good indication that it could be simplified.